datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaConnector.java

```diff
@@ -553,4 +570,12 @@ public AbstractKafkaBasedConnectorTask getConnectorTask() {
       return _task;
     }
   }
+
+  @Override
+  public List<BrooklinMetricInfo> getMetricInfos() {
+    List<BrooklinMetricInfo> metrics = new ArrayList<>();
+    metrics.add(new BrooklinMeterInfo(buildMetricName(_metricsPrefix + MetricsAware.KEY_REGEX + NUM_TASK_RESTARTS)));
```

Comment:
This should be `BrooklinCounterInfo` if the metric is a counter.

Reply:
@ahmedahamid instead of using buildMetricName() here, which prepends the class, can we instead do what other  similar metrics do? One example from `KafkaMirrorMakerConnectorTask`:

`    metrics.add(new BrooklinMeterInfo(generateMetricsPrefix(connectorName, CLASS_NAME) + MetricsAware.KEY_REGEX
        + TASK_LOCK_ACQUIRE_ERROR_RATE));`

Reply:
Discussed with @ahmedahamid offline, and we don't need a MetricsAware.KEY_REGEX here, since we don't have any dynamic element in the metric name. For the connector task metrics on the other hand, we do have the datastream or topic name as a dynamic part.

Reply:
@somandal discussed this with Sonam. We do not need Regex here as we do not have any variable parts here. But I made this a Gauge.